src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakIntegrator.h

```diff
@@ -568,6 +606,64 @@ namespace OpenMS
       const bool is_left_half
     ) const;
 
+    /**
+      @brief Given a saturated or cutoff peak, extract a training set to be used
+      with the gradient descent algorithm.
+
+      The algorithm tries to select a subset of points that best describe a non-saturated
+      peak. The decision of which points to skip is based on the derivatives between consecutive points.
+
+      It first selects all those points whose intensity is below a certain value (`intensity_threshold`).
```

Comment:
Use of an `intensity_threshold` maybe problematic if it is based on an absolute value because the absolute value is instrument dependent.